src/bluesearch/embedding_models.py

```diff
@@ -471,7 +296,11 @@ def compute_database_embeddings(connection, model, indices, batch_size=10):
     return final_embeddings, retrieved_indices
 
 
-def get_embedding_model(model_name_or_class, checkpoint_path=None, device=None):
+def get_embedding_model(
+    model_name_or_class: str,
+    checkpoint_path: Optional[Union[pathlib.Path, str, None]] = None,
```

Comment:
```suggestion
    checkpoint_path: Optional[Union[pathlib.Path, str]] = None,
```

Reply:
Hmm, I am not sure if I get this change, could you explain?

Reply:
Ahh, nice find!